crates/ruff/src/settings/line_width.rs

```diff
@@ -0,0 +1,205 @@
+use ruff_cache::CacheKey;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use unicode_width::UnicodeWidthChar;
+
+use ruff_macros::CacheKey;
+
+pub trait LineWidthState: Clone + Default + CacheKey {}
+
+#[derive(Clone, Copy, Debug, CacheKey)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct Length;
+
+impl Default for Length {
+    fn default() -> Self {
+        Self
+    }
+}
```

Comment:
You can derive `Default` automatically

```suggestion
#[derive(Clone, Copy, Debug, CacheKey, Default)]
#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
pub struct Length;
```